server/src/main/java/org/elasticsearch/action/admin/indices/forcemerge/TransportForceMergeAction.java

```diff
@@ -92,12 +93,16 @@ protected void shardOperation(
         ActionListener<TransportBroadcastByNodeAction.EmptyResult> listener
     ) {
         assert (task instanceof CancellableTask) == false; // TODO: add cancellation handling here once the task supports it
-        threadPool.executor(ThreadPool.Names.FORCE_MERGE).execute(ActionRunnable.supply(listener, () -> {
+        SubscribableListener.<IndexShard>newForked(l -> {
             IndexShard indexShard = indicesService.indexServiceSafe(shardRouting.shardId().getIndex())
                 .getShard(shardRouting.shardId().id());
-            indexShard.forceMerge(request);
-            return EmptyResult.INSTANCE;
-        }));
+            indexShard.ensureMutable(l.map(unused -> indexShard));
+        }).<EmptyResult>andThen((l, indexShard) -> {
```

Comment:
sadly we cannot use the variant of this method that takes an executor, as if the previous step has completed when the `andThen` is executed, it's executed in the calling thread :(

Reply:
But it's fine I guess as we're OK with a transport thread calling the ensureMutable, right?

Reply:
This action gets dispatched into the `MANAGEMENT` thread pool.

Reply:
And I meant that we have to dispatch into the `FORCE_MERGE` threadpool manually instead of relying on the `SubscribableListener` methods.